### ISSUE TITLE
(cf. #2204) Unreliable hub branch detection ?

### ISSUE THREAD

**Issue by buixor** [crowdsecurity/crowdsec#2205]
### What happened?

`cwversion.VersionStrip` now splits on `~` instead of `-`.
So it means that if crowdsec (`chooseHubBranch`) detects that crowdsec version is outdated, then it will try to use  the result of `cwversion.VersionStrip` to pick the branch. It will then lead to crowdsec trying to checkout a branch like `v1.5.0-debian-pragmatic` and so on,  which will *not* work.

### What did you expect to happen?

n/a

### How can we reproduce it (as minimally and precisely as possible)?

n/a

### Anything else we need to know?

n/a

### Crowdsec version

<details>

```console
$ cscli version
# paste output here
```

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Enabled collections and parsers

<details>

```console
$ cscli hub list -o raw
# paste output here
```

</details>

### Acquisition config

<details>
```console
# On Linux:
$ cat /etc/crowdsec/acquis.yaml /etc/crowdsec/acquis.d/*
# paste output here

# On Windows:
C:\> Get-Content C:\ProgramData\CrowdSec\config\acquis.yaml
# paste output here
</details>


### Config show

<details>

```console
$ cscli config show
# paste output here
```

</details>

### Prometheus metrics

<details>

```console
$ cscli metrics
# paste output here
```

</details>

### Related custom configs versions (if applicable) : notification plugins, custom scenarios, parsers etc.

<details>

</details>

**Comment by eguaj** [comment]
I also noticed this message after updating the Ubuntu .deb package from v1.4.6 to v1.5.0.

**Comment by blotus** [comment]
Fixed by #2210 and #2206.
